activemodel/lib/active_model/validator.rb

```diff
@@ -136,6 +136,7 @@ class EachValidator < Validator
     # +options+ reader, however the <tt>:attributes</tt> option will be removed
     # and instead be made available through the +attributes+ reader.
     def initialize(options)
+      options = options.dup
```

Comment:
Instead of fixing this just for `EachValidator`, what about fixing this for all validators passed to `validates_with`?  We could add a `dup` call to:

https://github.com/rails/rails/blob/e20e541ff3ea3732359334d47d256830450b9008/activemodel/lib/active_model/validations/with.rb#L86

and to:

https://github.com/rails/rails/blob/e20e541ff3ea3732359334d47d256830450b9008/activemodel/lib/active_model/validations/with.rb#L142

Reply:
In my point of view this would be a breaking change, too. 
The docs for  `ActiveModel::EachValidator` says that "however the :attributes option will be removed".

Reply:
| I'm having trouble imagining what this would break. Could you provide an example?

Never mind, I have some examples, but my implementation will breaks these examples too:

The first example is a user who read the docs implemented his own validator and wrote a test that checks that the key is removed. 

Next it's possible that some one wrote a custom validator and uses keyword arguments, maybe this valitator relys on the `EachValidator`  that this validator removes the `attributes` key or it will fail.

or code like this will begin to fail:

```ruby
   def my_validations
      opts = { attributes: [:name], allow_nil: true }
     validates_with ActiveRecord::Validations::UniquenessValidator, opts
     my_things **opts
  end
```

Reply:
| If the dup is called in validates_with, EachValidator would behave the same as it does now, so such a custom validator would also behave the same. Or am I misunderstanding?

In the current implementation, when you call a `EachValidator` first and the second is a custom validator, then the second will get a Hash passed without a `:attributes` key. When did make the mistake and expect keyword arguments  instead of a Hash, then it will fail as soon as you put the Hash within `validates_with`, but like I said, this will also fail with my implementation, so this introduce a breaking change too.

| In that case I believe the following would work: ...

yes, this would work ... but you expect users to change their code, just because you keep your promise any longer. 

| If Rails changed the implementation of UniquenessValidator to not use EachValidator, I don't think we would consider it a breaking change.

I would call it a breaking change. With inheriting from EachValidator you make a promise for a certain interface. You promise that it will behave like a EachValidator.

Reply:
At the end, this PR/issue has been quiet for 4 month, I don't think that this is realy an issue that some one needs to be fixed.